defi_repertoire/utils.py

```diff
@@ -8,6 +11,11 @@ def flatten(matrix):
 def uniqBy(l, attr):
     seen = {}
     for obj in l:
-        if obj[attr] not in seen:
-            seen[obj[attr]] = obj
+        if type(obj) is dict:
+            if obj[attr] not in seen:
+                seen[obj[attr]] = obj
+        else:
+            if getattr(obj, attr) not in seen:
+                seen[getattr(obj, attr)] = obj
```

Comment:
A bit more Pythonic, safe (`is` is not great to use), and readable

```suggestion
key = obj[attr] if isinstance(obj, dict) else getattr(obj, attr)

if key not in seen:
    seen[key] = obj
```

Reply:
needs indentation fix from suggestion